client/windows/CCastleInterface.cpp

```diff
@@ -874,7 +876,16 @@ void CCastleBuildings::enterMagesGuild()
 
 	if(hero && !hero->hasSpellbook()) //hero doesn't have spellbok
 	{
-		if(LOCPLINT->cb->getResourceAmount(Res::GOLD) < 500) //not enough gold to buy spellbook
+		const StartInfo *si = LOCPLINT->cb->getStartInfo();
+		// it would be nice to find a way to move this hack to config/mapOverrides.json
+		if (si && si->campState && si->campState->camp &&               // We're in campaign,
+			(si->campState->camp->header.filename == "DATA/YOG.H3C") && // which is "Birth of a Barbarian",
+			(hero->subID == 45) && (hero->name == "Yog"))               // and the hero is Yog (based on Solmyr)

```

Comment:
You should not check hero name, it may be localized

Reply:
Riiight. I was only unsure whether the player can buy an actual Solmyr, but now I remember heroes can't be duplicate.